flow/display_list_interpreter.h

```diff
@@ -290,31 +293,32 @@ class DisplayListInterpreter {
       return rrect;
     }
 
-    uint32_t GetIntList(uint32_t** int_ptr) {
+    uint32_t GetIntList(uint32_t** uint32_list_ptr) {
       uint32_t len = GetUint32();
-      *int_ptr = (uint32_t*)&*data;
+      *uint32_list_ptr = reinterpret_cast<uint32_t*>(&*data);
       skipData(len);
       return len;
     }
 
-    uint32_t GetFloatList(SkScalar** flt_ptr) {
+    uint32_t GetFloatList(SkScalar** scalar_list_ptr) {
       uint32_t len = GetUint32();
-      *flt_ptr = &*data;
+      *scalar_list_ptr = reinterpret_cast<SkScalar*>(&*data);
       skipData(len);
       return len;
     }
 
-    const sk_sp<SkColorFilter> GetColorFilter() {
+    const sk_sp<SkColorFilter> GetColorFilterRef() {
       return (refs++)->colorFilter;
     }
-    const sk_sp<SkImageFilter> GetImageFilter() {
+    const sk_sp<SkImageFilter> GetImageFilterRef() {
       return (refs++)->imageFilter;
     }
-    const sk_sp<SkImage> GetImage() { return (refs++)->image; }
-    const sk_sp<SkVertices> GetVertices() { return (refs++)->vertices; }
-    const sk_sp<SkShader> GetShader() { return (refs++)->shader; }
-    const sk_sp<SkPicture> GetSkPicture() { return (refs++)->picture; }
-    const DisplayListData GetDisplayList() { return (refs++)->displayList; }
+    const sk_sp<SkShader> GetShaderRef() { return (refs++)->shader; }
+
+    const SkImage* GetImage() { return (refs++)->image.get(); }
+    const SkVertices* GetVertices() { return (refs++)->vertices.get(); }
+    const SkPicture* GetSkPicture() { return (refs++)->picture.get(); }
```

Comment:
I would find it helpful if we documented why we're returning a raw pointer here and what the ownership expectations are. It seems like we're saying that at this pointer is only good until the iterator is advanced again?

I'm a little worried that this will be brittle to refactoring and testing. It would be easier to reason about this API and review future refactoring of it if it just had a clear ownership model - but at least if it's documented we won't have to re-read it to figure out what's going on.

Reply:
I'm waffling on whether to return Refs or pointers here. For most cases, we don't need to take ownership of these values because they are used in a single op that goes out of scope.

Note that I still return Refs for attributes because SkPaint takes refs (for obvious reasons).